### ISSUE TITLE
Progress bar crashes on terminal resize

### ISSUE THREAD

**Issue by squarism** [peter-murach/tty-progressbar#11]
Running the progress bar from the example docs.  Resize the terminal while the progress is going.
Tried iTerm and OSX terminal, same result.

```
~/.rubies/ruby-2.1.6/lib/ruby/2.1.0/monitor.rb:246:in `mon_check_owner': current thread not owner (ThreadError)
	from ~/.rubies/ruby-2.1.6/lib/ruby/2.1.0/monitor.rb:195:in `mon_exit'
	from ~/.rubies/ruby-2.1.6/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:143:in `ensure in require'
	from ~/.rubies/ruby-2.1.6/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:143:in `require'
	from ~/.gem/ruby/2.1.6/gems/tty-screen-0.5.0/lib/tty/screen/size.rb:47:in `try_io_console'
	from ~/.gem/ruby/2.1.6/gems/tty-screen-0.5.0/lib/tty/screen/size.rb:38:in `from_io_console'
	from ~/.gem/ruby/2.1.6/gems/tty-screen-0.5.0/lib/tty/screen/size.rb:22:in `size'
	from ~/.gem/ruby/2.1.6/gems/tty-screen-0.5.0/lib/tty/screen.rb:45:in `size'
	from ~/.gem/ruby/2.1.6/gems/tty-screen-0.5.0/lib/tty/screen.rb:26:in `size'
	from ~/.gem/ruby/2.1.6/gems/tty-screen-0.5.0/lib/tty/screen.rb:31:in `width'
	from ~/.gem/ruby/2.1.6/gems/tty-progressbar-0.8.0/lib/tty/progressbar.rb:266:in `max_columns'
	from ~/.gem/ruby/2.1.6/gems/tty-progressbar-0.8.0/lib/tty/progressbar.rb:289:in `block in register_callbacks'
	from tty_time.rb:5:in `call'
	from tty_time.rb:5:in `sleep'
	from tty_time.rb:5:in `block in <main>'
	from tty_time.rb:4:in `times'
	from tty_time.rb:4:in `<main>'
```

I was poking around to find more behavior but ran out of time.

**Comment by peter-murach** [comment]
I've been using the progress bar a lot but haven't been resizing the screen. Thanks for reporting. Would you find time to investigate further?

**Comment by squarism** [comment]
I think I have made progress on this with #12 .  Of course it's up to you if you like the change.  :sunflower:

**Comment by peter-murach** [comment]
I've released `v0.8.1` with a patch. I've decided to only trap interrupt signal, trapping exit would swallow genuine exceptions. The `resize` has been simplified and tested to properly work inside threads and actually resize the progress bar on terminal size change. Please let me know if this works for you.

**Comment by squarism** [comment]
Nice!  Seems to work and the change makes sense.  Thanks for merging a bit of it.  Glad to help.  :sunflower:
